state_test.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/ory/dockertest"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupPostgresql(t *testing.T) (*sql.DB, func()) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		t.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	resource, err := pool.Run("postgres", "latest", []string{"POSTGRES_HOST_AUTH_METHOD=trust"})
+	if err != nil {
+		t.Fatalf("Could not start resource: %s", err)
+	}
+
+	endpoint := fmt.Sprintf("postgres://postgres@localhost:%s/postgres?sslmode=disable", resource.GetPort("5432/tcp"))
+
+	// wait for the db to be running
+	var db *sql.DB
+	if err := pool.Retry(func() error {
+		db, err = sql.Open("postgres", endpoint)
+		if err != nil {
+			return err
+		}
+		return db.Ping()
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup := func() {
+		if err := pool.Purge(resource); err != nil {
+			t.Fatalf("Could not purge resource: %s", err)
+		}
+	}
+	return db, cleanup
+}
+
+func execSQLFile(t *testing.T, path string, tx *sql.Tx) error {
+	c, ioErr := ioutil.ReadFile(path)
+	if ioErr != nil {
+		t.Fatal("Cannot Read SQL file.")
+	}
+	sql := string(c)
+
+	_, err := tx.Exec(sql)
+	if err != nil {
+		return err
+	}
+
+	t.Log(sql)
+	return nil
+}
+
+func setupPostgresqlTables(t *testing.T, db *sql.DB) error {
+
+	tx, err := db.Begin()
+	if err != nil {
+		log.Info(err)
+	}
+
+	if err := execSQLFile(t, "./migrations/01-block.sql", tx); err != nil {
```

Comment:
Where are you creating the tables in the production db? Move this logic to a function:
```
func (s *State) migrate() error {

}
```
that you run every time that State is created, not only during tests.